tests/src/test/scala/whisk/core/container/test/ContainerUtilsTests.scala

```diff
@@ -0,0 +1,147 @@
+
+    def writeLogFile(fixture: FixtureParam, content: String): Long = {
+        fixture.writer.write(content)
+        fixture.writer.flush()
+        fixture.file.length
+    }
+
+    it should "tolerate an empty log file" in { fixture =>
+        val logText = ""
+        val size = writeLogFile(fixture, logText)
+
+        val buffer = fixture.cu.getDockerLogContent(containerHash = ContainerHash.fromString("0123"), start = 0, end = size, mounted = false)
+        val logContent = new String(buffer.array, buffer.arrayOffset, buffer.position, StandardCharsets.UTF_8)
+
+        logContent shouldBe logText
+        stream.size() shouldBe 0
+    }
+
+    it should "read a full log file" in { fixture =>
+        val logText = "text"
+        val size = writeLogFile(fixture, logText)
+
+        val buffer = fixture.cu.getDockerLogContent(containerHash = ContainerHash.fromString("0123"), start = 0, end = size, mounted = false)
+        val logContent = new String(buffer.array, buffer.arrayOffset, buffer.position, StandardCharsets.UTF_8)
+
+        logContent shouldBe logText
+        stream.size() shouldBe 0
+    }
+
+    it should "read a log file portion" in { fixture =>
+        val count = 4
+        val logText = (1 to count).map(i => s"Line ${i}\n").mkString("")
+        val size = writeLogFile(fixture, logText)
+        val quarter = count / 4
+        val qlen = logText.length / 4 // length of a quarter
+        val from = qlen // start reading from second quarter
+        val to = from + 2 * qlen // read second and third quarter
+        val expectedText = logText.substring(from, to)
+
+        val buffer = fixture.cu.getDockerLogContent(containerHash = ContainerHash.fromString("0123"), start = from, end = to, mounted = false)
+        val logContent = new String(buffer.array, buffer.arrayOffset, buffer.position, StandardCharsets.UTF_8)
+
+        logContent shouldBe expectedText
+        stream.size() shouldBe 0
+    }
+
+    it should "tolerate premature end of log file" in { fixture =>
+        val logText = (1 to 2).map(i => s"Line ${i}\n").mkString("")
+        val size = writeLogFile(fixture, logText)
+        val to = 2 * size
+
+        val buffer = fixture.cu.getDockerLogContent(containerHash = ContainerHash.fromString("0123"), start = 0, end = to, mounted = false)
+        val logContent = new String(buffer.array, buffer.arrayOffset, buffer.position, StandardCharsets.UTF_8)
+
+        logContent shouldBe logText
+        stream.size() shouldBe 0
+    }
+
+    it should "provide an empty result on failure" in { fixture =>
+        fixture.cu.logFile = new File("/nonsense")
+
+        val buffer = fixture.cu.getDockerLogContent(containerHash = ContainerHash.fromString("0123"), start = 0, end = 1, mounted = false)
+
+        buffer.capacity() shouldBe 0
+        logLines.head should include("getDockerLogContent failed")
+    }
+}
+
+class ContainerUtilsTester(var logFile: File)(implicit val logging: Logging) extends ContainerUtils {
```

Comment:
I understand... When keeping the fixture approach, test `provide an empty result on failure` will be invoked with a fixture that isn't really used...

At the same time, said test overrides the `logFile` to a file that doesn't exist to provoke a failure.